### PR TITLE
Adding a CLI

### LINKED ISSUE #13: cli tool
is it possible to include a command line interface for it?
like install it globally and run it from project root as in npm-check-updates package

### PR DESCRIPTION
**Pull request by TheSavior**
Adding a CLI

Resolves #13.

I'm not really sure how to test this following the pattern of the rest of the tests. Take a look, let me know what you think.

### CODE REVIEW

**Comment by TheSavior** [comment]
@mzgol Any update on this?

File: README.md [hunk 1]
```diff
@@ -18,13 +18,27 @@ npm install check-dependencies --save-dev
...
+### CLI
+
+```bash
+$ check-dependencies
+```
+
+The options listed below can be passed to the CLI as well.
+
+```bash
+$ check-dependencies --install --verbose
```
Comment (mzgol) [inline README.md]: What about parameters that accept options?

File: test/spec.js [hunk 2]
```diff
@@ -725,4 +726,69 @@ describe('checkDependencies', function () {
             });
         }
     });
+
+    describe('cli', function () {
```
Comment (mzgol) [inline test/spec.js]: I'd like to have at least 2 test (success & failure) that use regular spawned process runnin `./bin/cli.js` with proper parameters to be sure that not only the "CLI API" works but the command itself.

**Comment by mzgol** [comment]
@TheSavior I was on vacation, no worries; 6 days is not a lot. ;)

I like it! I left some comments.

**Comment by TheSavior** [comment]
It looks like I'm not going to be able to prioritize finishing this up in the forseeable future. If anyone is willing to take over this PR I'm sure we'd all be quite greatful!

**Comment by TheSavior** [comment]
I was able to add two tests for the CLI that spawn it for different packages. It should be easy enough for someone to copy that format to test with parameters.

I'm pretty sure that the CLI as it stands in this PR won't handle parameters that aren't just true or false. Any ideas on how to do this without just duplicating all of the configuration stuff in the cli.js file? I was trying to be able to just pass an object through to the main object so it will handle changes transparently.

File: bin/cli.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+var minimist = require('minimist');
+var checkDependencies = require('../');
+
+var argv = minimist(process.argv.slice(2));
+
+var Cli = {
+  reporter: function (result) {
+    result.error.forEach(function (dep) {
+      console.log(dep);
```
Comment (mzgol) [inline bin/cli.js]: Why `console.log` and not `console.error`? Also, it'd be better to change the name `dep` to `msg` or sth like that; after all, these are messages you're logging, not dependencies (i.e. not just dependencies' names are logged but complete messages).

**Comment by mzgol** [comment]
> I'm pretty sure that the CLI as it stands in this PR won't handle parameters that aren't just true or false. Any ideas on how to do this without just duplicating all of the configuration stuff in the cli.js file?

Can't you treat every parameter as string and run `JSON.parse` on those that are expected to be a richer structure than just a string? This won't be as flexible as using JS directly but almost every parameter should be possible to present like that. Example usage:
```bash
check-dependencies --verbose --package-manager="npm" --scope-list="['dependencies', 'devDependencies']"
```

Note that I wrote `--package-manager`, not `--packageManager` as that's usually the format used in various CLIs. I'm not sure if minimist is handling that automatically or not but it would be good to have a test for that.

File: test/spec.js [hunk 4]
```diff
@@ -725,4 +728,106 @@ describe('checkDependencies', function () {
...
+                    assert.strictEqual(output, null);
+                    done();
+                });
+            });
+
+            it('should be failure on a non ok package', function (done) {
+                var packageRoot = path.join(fixturesRoot, 'latest-not-ok');
+
+                var child = childProcess.spawn(process.execPath, [cliPath], {
+                    cwd: packageRoot,
```
Comment (mzgol) [inline test/spec.js]: It would be cool if at least one test was run with parameters.

**Comment by mzgol** [comment]
Thanks, I've landed it with minor corrections, I also added support for non-boolean parameters.

**Comment by TheSavior** [comment]
Thank you for finishing this up!

**Pull request closed without merging**